### PR TITLE
microcosm-sagemaker>=0.7.0

### PR DESCRIPTION
**Pull request by pokey**
Depends on https://github.com/globality-corp/terraform-services/pull/352

### CODE REVIEW

**Review by piercefreeman — APPROVED** [review]

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/app_hooks/train/app.py [hunk 1]
```diff
@@ -9,7 +9,10 @@
 from microcosm_sagemaker.loaders import train_conventions_loader
 
 import {{ cookiecutter.package_name }}.bundles  # noqa: 401
-from {{ cookiecutter.package_name }}.app_hooks.train.config import load_default_config
+import {{ cookiecutter.package_name }}.evaluations  # noqa: 401
+from {{cookiecutter.package_name}}.app_hooks.train.config import (
```
Comment (GabeLambda) [inline {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/app_hooks/train/app.py]: Why the change in spacing?
It's also inconsistent with the line above.

**Comment by jameschung2019** [comment]
Hey there, please close this PR or merge it. It will be auto-closed in 48 hours if not acted upon. If you decide to merge it, comment and take ownership to coordinate, schedule, and get it done. Thanks!

**Comment by jameschung2019** [comment]
This PR is being closed per CTO leadership's decision.  All PRs older than 1/1/2020 is included.

**Review by salmanmashayekh — APPROVED** [review]
LGTM. 
Not adding pipeline testing?

File: setup.cfg [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
 [isort]
 combine_as_imports = True
 enforce_white_space = True
 force_grid_wrap = 4
 include_trailing_comma = True
-known_first_party = cookiecutter_microcosm_sagemaker
+known_first_party = papaya_extractor
 known_standard_library = dataclasses,pkg_resources
 known_third_party = alembic,allennlp,boto3,click,hamcrest,joblib,matplotlib,microcosm,microcosm_sagemaker,networkx,node2vec,nose,numpy,pandas,parameterized,seaborn,six,sklearn,taxonomies,tf,torch,tqdm,unidecode
```
Comment (marcglobality) [inline setup.cfg]: I have a longer list if you want to include it

old:
known_third_party = alembic,allennlp,boto3,click,hamcrest,joblib,matplotlib,microcosm,microcosm_sagemaker,networkx,node2vec,nose,numpy,pandas,parameterized,seaborn,six,sklearn,taxonomies,tf,torch,tqdm,unidecode

updated:
known_third_party = alembic,allennlp,boto3,bs4,click,dragnet,hamcrest,joblib,lxml,marshmallow,matplotlib,microcosm,microcosm_flask, microcosm_sagemaker,microcosm_secretsmanager,microcosm_logging,networkx,node2vec,nose,numpy,pandas,parameterized,postal,requests,seaborn,six,sklearn,taxonomies,tf,tldextract,torch,torchtext,tqdm,unidecode,wandb

Comment (marcglobality) [reply setup.cfg]: btw, isn't there any way to tell isort that everything is third party unless is first party?

File: {{cookiecutter.repository_name}}/README.md [hunk 3]
```diff
@@ -5,7 +5,8 @@
```
Comment (marcglobality) [inline {{cookiecutter.repository_name}}/README.md:5]: I added some extras to the README in my PRs, maybe we want to include (some of) that? see e.g. https://github.com/globality-corp/web-annotation-extractor/pull/3/files#diff-04c6e90faac2675aa89e2176d2eec7d8

File: {{cookiecutter.repository_name}}/setup.cfg [hunk 4]
```diff
@@ -15,8 +15,8 @@ force_grid_wrap=4
 include_trailing_comma=True
 known_first_party={{ cookiecutter.package_name }}
 known_standard_library=dataclasses,pkg_resources
-known_third_party=six,hamcrest,mock,nose,parameterized,click,microcosm_sagemaker,microcosm
-line_length=79
+known_third_party=six,hamcrest,mock,nose,parameterized,click,microcosm_sagemaker,microcosm,microcosm_logging,microcosm_flask,marshmallow
```
Comment (marcglobality) [inline {{cookiecutter.repository_name}}/setup.cfg:18]: same as https://github.com/globality-corp/cookiecutter-microcosm-sagemaker/pull/21/files?file-filters%5B%5D=.cfg&file-filters%5B%5D=.json&file-filters%5B%5D=.md&file-filters%5B%5D=.py&file-filters%5B%5D=.sh&file-filters%5B%5D=.txt#r424945578

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/app_hooks/evaluate/app.py [hunk 5]
```diff
@@ -10,7 +10,7 @@
 
 import {{ cookiecutter.package_name }}.bundles  # noqa: 401
 import {{ cookiecutter.package_name }}.evaluations  # noqa: 401
-from {{ cookiecutter.package_name }}.app_hooks.evaluate.config import load_default_config
+from {{cookiecutter.package_name}}.app_hooks.evaluate.config import load_default_config
```
Comment (marcglobality) [inline {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/app_hooks/evaluate/app.py:13]: I don't see why spaces are required in 11,12 but not in 13. Same in other places along the repo

**Comment by marcglobality** [comment]
> LGTM.
> Not adding pipeline testing?

That'd be great! If so, please remember that new version of sagemaker has `PipelineHarness` defined

File: {{cookiecutter.repository_name}}/setup.py [hunk 6]
```diff
@@ -18,9 +18,9 @@
     install_requires=[
         "click>=7.0",
         "microcosm-logging>=1.3.0",
-        "microcosm-sagemaker>=0.4.1",
+        "microcosm-sagemaker>=0.8.2",
         "microcosm-secretsmanager>=1.1.0",
-        "microcosm>=2.12.0",
+        "microcosm>=2.12.2",
```
Comment (marcglobality) [inline {{cookiecutter.repository_name}}/setup.py:23]: maybe add the lint extras in this file?

```
"lint": [
            "flake8>=3.6.0",
            "flake8-isort>=2.6.0",
            "flake8-print>=3.1.0",
            "flake8-logging-format",
        ],
```

**Review by andrewplumb — APPROVED** [review]

File: .circleci/config.yml [hunk 7]
```diff
@@ -64,11 +64,37 @@ jobs:
...
+              --build-arg BUILD_NUM=$CIRCLE_BUILD_NUM \
+              --build-arg SHA1=$CIRCLE_SHA1 \
+              --build-arg EXTRA_INDEX_URL=$EXTRA_INDEX_URL \
+              --build-arg JFROG_AUTH=$JFROG_AUTH .
+
+            # Push from local machine (requires AWS ECR permissions):
+            # eval $(botoenv -p development)
+
+            eval $(aws ecr get-login --no-include-email)
+            docker push $DOCKER_IMAGE_SHA_TAG
```
Comment (pokey) [inline .circleci/config.yml]: So is all of this code handwritten, autogenerated, or copied and modified from autogenerated?

File: {{cookiecutter.repository_name}}/.globality/build.json [hunk 8]
```diff
@@ -1,5 +1,13 @@
 {
   "params": {
+    "entrypoint": {
+      "pre_service_commands": [
+        "pip install awscli",
+        "aws s3 cp ${MODEL_LOCATION} .",
+        "mkdir -p /opt/ml/model",
+        "tar -zxf model.tar.gz -C /opt/ml/model"
+      ]
+    },
```
Comment (pokey) [inline {{cookiecutter.repository_name}}/.globality/build.json]: I believe this stuff is old.  I think you can find the latest here: https://github.com/globality-corp/experimental-matcher/blob/master/.globality/build.json.  @pankajghosh anything else that needs updating to latest CD setup?

**Review by pokey — COMMENTED** [review]
LGTM with a couple minor comments.  (can't approve because technically it's my PR)

File: {{cookiecutter.repository_name}}/.globality/build.json (hunk 8 above)
Comment (pankajghosh) [reply {{cookiecutter.repository_name}}/.globality/build.json]: yup, I would use existing AI service build.json as a template for this. Few things:
1. No need for entrypoint section anymore.
2. environment section should only have a "demo" section(since we do not autodeploy to devx)
3. type should be ai-model
4. update_commands needs to be fixed to use v2 (new CD command)

**Comment by pankajghosh** [comment]
@pokey Also, I would suggest adding a data_files folder with a sample or empty datasets.json file.

File: .circleci/config.yml (hunk 7 above)
Comment (piercefreeman) [reply .circleci/config.yml]: All autogenerated @pokey based on the new globality-build file in the root of this repo
